spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServerConfigDataLoader.java

```diff
@@ -319,12 +319,21 @@ protected Environment getRemoteEnvironment(ConfigDataLoaderContext context, Conf
 				HttpHeaders headers = new HttpHeaders();
 				headers.setAccept(acceptHeader);
 				requestTemplateFactory.addAuthorizationToken(headers, username, password);
-				if (StringUtils.hasText(token)) {
-					headers.add(TOKEN_HEADER, token);
-				}
 				if (StringUtils.hasText(state) && properties.isSendState()) {
 					headers.add(STATE_HEADER, state);
 				}
+				if (StringUtils.hasText(properties.getTokenUri()) && !properties.getHeaders().isEmpty()) {
+					List<ClientHttpRequestInterceptor> interceptors = List
+							.of(new ConfigClientRequestTemplateFactory.GenericRequestHeaderInterceptor(
+									properties.getHeaders()));
+					restTemplate.setInterceptors(interceptors);
+				}
+				else {
+					requestTemplateFactory.addAuthorizationToken(headers, username, password);
```

Comment:
What happens if this token expires?

Reply:
In general, the Spring Security Framework handles refreshing of the token. I am still trying to set up a test for the actuator monitor endpoint to see if the security framework will append the token to the config server request to pull new updates. In theory it should work since the Spring Context is fully boot strapped.

Reply:
OK please let us know when you have tested that

Reply:
Given Spring Cloud Bus enabled. Testing webhook and actuator endpoints via Postman with OAuth header.

Client: 
- actuator/refresh (test case for no cloud bus) Bearer token appended on config server call after initial load and after token expiration. 
- actuator/busrefresh Bearer token appended on config server call after initial load and after token expiration.

Config Server: 
- actuator/monitor queued actuator/busrefresh Client listener picked up refresh and Bearer token appended to config server request.

Reply:
@ryanjbaxter @spencergibb 
If I need to bring in a JWT Decoder, how do you package it so it is not forced onto the client project?
```
		<dependency>
			<groupId>com.auth0</groupId>
			<artifactId>java-jwt</artifactId>
			<version>4.4.0</version>
		</dependency>
```

When I added this to the client in order to refresh the token, I was forced to put this dependency in my Client.